src/database_observer.py

```diff
@@ -45,22 +44,52 @@ def __init__(self, charm: CharmBase, relation_name: str):
 
     def _on_database_created(self, _: DatabaseCreatedEvent) -> None:
         """Handle database created."""
-        self._charm.reconcile()  # type: ignore
+        self._charm.reconcile()
 
     def _on_endpoints_changed(self, _: DatabaseEndpointsChangedEvent) -> None:
         """Handle endpoints changed."""
-        self._charm.reconcile()  # type: ignore
+        self._charm.reconcile()
 
     def get_db(self) -> typing.Optional[DatasourcePostgreSQL]:
         """Return a postgresql datasource model.
 
         Returns:
             DatasourcePostgreSQL: The datasource model.
         """
-        # not using get_relation due this issue
-        # https://github.com/canonical/operator/issues/1153
-        if not self.model.relations.get(self.database.relation_name):
+        relation_data = list(
+            self.database.fetch_relation_data(
+                fields=["uris", "endpoints", "username", "password", "database"]
+            ).values()
+        )
+
+        if not relation_data:
+            return None
+
+        # There can be only one database integrated at a time
+        # with the same interface name. See: metadata.yaml
+        data = relation_data[0]
+
+        # Check that the relation data is well formed according to the following json_schema:
+        # https://github.com/canonical/charm-relation-interfaces/blob/main/interfaces/mysql_client/v0/schemas/provider.json
+        if not all(data.get(key) for key in ("endpoints", "username", "password")):
             return None
 
-        relation = self.model.get_relation(self.relation_name)
-        return DatasourcePostgreSQL.from_relation(relation)
+        database_name = data.get("database", self.database.database)
+        endpoint = data["endpoints"].split(",")[0]
+        user = data["username"]
+        password = data["password"]
+        host, port = endpoint.split(":")
+
+        if "uris" in data:
+            uri = data["uris"].split(",")[0]
```

Comment:
Leaving it since IRC bridge expects this format
https://github.com/matrix-org/matrix-appservice-irc/blob/062f3ace8b08e86e9f63c004ffdb85942af1b94e/config.sample.yaml#L703